tools/format.ts

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env deno --allow-run
+import {
+  cwd,
+  exit,
+  platform,
+  Process,
+  ProcessStatus,
+  readAll,
+  readDirSync,
+  run,
+  RunOptions
+} from "deno";
+
+// TODO: we might need a fix_symlinks() equivalent (eg in third_party.py).
+
+type ProcessOptions = [string, RunOptions];
+
+function runp(runOpts: RunOptions) {
+  return run(
+    Object.assign(
+      {
+        stdout: "piped",
+        stderr: "piped"
+      },
+      runOpts
+    )
+  );
+}
+
+function pathJoin(...args: string[]): string {
+  return args.join("/");
```

Comment:
I wasn't really sure about the limitations of deno + windows regarding joining paths etc, I'm used to just defaulting to use `path.join()` in nodejs but I've never really needed to worry about it too much. figured I should put this here as a placeholder and ask first.